code/modules/chemistry/tools/dispensers.dm

```diff
@@ -360,6 +360,15 @@
 		else if (ismob(usr))
 			add_fingerprint(usr, TRUE)
 
+	smash()
+		..()
+		icon_state = "weldtank1"
+
+	ex_act(severity)
+		..()
+		icon_state = "weldtank1" //to ensure that a weldertank's always going to be updated by their own explosion
```

Comment:
I think naming the icon state something like `weldtank-broken` or `weldtank-exploded` would be more readable than `weldtank1`.